capdl-loader-app/src/main.c

```diff
@@ -2113,8 +2114,9 @@ static char ALIGN(PAGE_SIZE_4K) debug_libc_morecore_area[PAGE_SIZE_4K];
 static void CONSTRUCTOR(MUSLCSYS_WITH_VSYSCALL_PRIORITY) init_bootinfo(void)
 {
     /* Init memory area for musl. */
-    morecore_area = debug_libc_morecore_area;
-    morecore_size = sizeof(debug_libc_morecore_area);
+    sel4muslcsys_setup_morecore_region(
```

Comment:
This function doesn't exist yet right?

Reply:
True, this would be a new function to guard the setup, the variables become private then and we can ensure things like proper alignment more easily. I've kept this as draft because there are still a few thing I need to understand better how this is used.